src/livestatus.c

```diff
@@ -0,0 +1,483 @@
+  "neb_callbacks "                                                             \
+  "neb_callbacks_rate "                                                        \
+  "requests "                                                                  \
+  "requests_rate "                                                             \
+  "service_checks "                                                            \
+  "service_checks_rate"
+
+#define LIVESTATUS_EXPECTED_FIELDS_RESP_NB 19
+
+struct livestatus_status_s {
+  int cached_log_messages;
+  lcounter_t connections;
+  double connections_rate;
+  lcounter_t forks;
+  double forks_rate;
+  lcounter_t host_checks;
+  double host_checks_rate;
+  lcounter_t livecheck_overflows;
+  double livecheck_overflows_rate;
+  lcounter_t livechecks;
+  double livechecks_rate;
+  lcounter_t log_messages;
+  double log_messages_rate;
+  lcounter_t neb_callbacks;
+  double neb_callbacks_rate;
+  lcounter_t requests;
+  double requests_rate;
+  lcounter_t service_checks;
+  double service_checks_rate;
+};
+typedef struct livestatus_status_s livestatus_status_t;
+
+#define LIVESTATUS_DEFAULT_SOCKET_FILE "/var/cache/naemon/live"
+
+static livestatus_t livestatus_obj = {
+    .socket_file = NULL,
+    .max_retry = 20,
+    .backoff_sec = 1,
+};
+
+livestatus_status_t c_to_livestatus_status(const char **fields) {
+  livestatus_status_t status = {
+      .cached_log_messages = atoi(fields[0]),
+      .connections = strtoull(fields[1], NULL, 10),
+      .connections_rate = strtod(fields[2], NULL),
+      .forks = strtoull(fields[3], NULL, 10),
+      .forks_rate = strtod(fields[4], NULL),
+      .host_checks = strtoull(fields[5], NULL, 10),
+      .host_checks_rate = strtod(fields[6], NULL),
+      .livecheck_overflows = strtoull(fields[7], NULL, 10),
+      .livecheck_overflows_rate = strtod(fields[8], NULL),
+      .livechecks = strtoull(fields[9], NULL, 10),
+      .livechecks_rate = strtod(fields[10], NULL),
+      .log_messages = strtoull(fields[11], NULL, 10),
+      .log_messages_rate = strtod(fields[12], NULL),
+      .neb_callbacks = strtoull(fields[13], NULL, 10),
+      .neb_callbacks_rate = strtod(fields[14], NULL),
+      .requests = strtoull(fields[15], NULL, 10),
+      .requests_rate = strtod(fields[16], NULL),
+      .service_checks = strtoull(fields[17], NULL, 10),
+      .service_checks_rate = strtod(fields[18], NULL),
+  };
+
+  return status;
+} /* livestatus_status_t c_to_livestatus_status */
+
+// Borrowed from https://stackoverflow.com/a/29378380
+static long ls_strtol_subrange(const char *s, char **endptr, int base, long min,
+                               long max) {
+  long y = strtol(s, endptr, base);
+  if (y > max) {
+    errno = ERANGE;
+    return max;
+  }
+  if (y < min) {
+    errno = ERANGE;
+    return min;
+  }
+  return y;
+} /* static long ls_strtol_subrange */
+
+// Borrowed from https://stackoverflow.com/a/29378380
+static int ls_strtoi(const char *s, char **endptr, int base) {
```

Comment:
Why is this restricting us to the `int` (usually 32 bit, but that's not guaranteed) range? The downstream code is using 64 bit counters consistently, so why reject counter values despite being able to handle them?